x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/action/TransportXPackInfoActionTests.java

```diff
@@ -43,6 +43,7 @@
 
 public class TransportXPackInfoActionTests extends ESTestCase {
 
+    @SuppressWarnings("unchecked")
```

Comment:
The other option here, rather than suppressing the warning for the whole method, is to define a helper like so:

```java
    @SuppressWarnings("unchecked")
    private <T> ActionListener<T> anyActionListener() {
        return any(ActionListener.class);
    }
```

then replace `any(ActionListener.class)` with `anyActionListener()`. It's slightly annoying to have to do this, but it's an easy way to limit the scope of the suppressions.

Reply:
That's a great suggestion! Unfortunately, that would still require adding suppression since the method does castings like  (ActionListener<XPackInfoFeatureResponse>)answer.getArguments()[2] on line 58. So, keeping this as is.

Reply:
Plus it would require adding helpers for eq on line 57 and any(ActionRequest) which could get a bit gawky, but happy to change this if you feel otherwise.